templates/vhost/_ssl.epp

```diff
@@ -0,0 +1,68 @@
+<% if $ssl { -%>
```

Comment:
Like `_setenv.epp`, I'd prefer to keep the `if` in the Puppet code and avoid rendering the whole template.